rfcs/text/i18n.md

```diff
@@ -0,0 +1,298 @@
+
+We will also deprecate all [`i18n` helpers](https://github.com/Sage/carbon/blob/a59afc80fd7ac954b58da5cd9ef278239b5756c0/src/utils/helpers/i18n/i18n.js). Continuing to support these is beyond the scope of a component library and is better handled by a dedicated library such as [`numeral.js`](http://numeraljs.com/).
+Removing these will further reduce the complexity of this project and allow for more advanced configurations by our consumers.
+
+Finally, each translation will be overridable by our consumers by providing a string prop to the component. This string 
+be also be provided by `react-i18next`.
+
+[1]: https://react.i18next.com/latest/usetranslation-hook
+[2]: https://www.i18next.com/overview/api#t
+
+# Motivation
+
+We currently only ship english translations. Each time a product launches in a non-english market they have to set up
+translations, these are typically non-contextual and will be the same in each product.
+
+Having `i18n-js` as a peer-dependency means that we're encouraging the use of this, unintentionally, for content translations.
+The current recommended setup is `i18next`, so consumers will have to configure both libraries, increasing the bundle size and reducing performance.
+
+# Detailed design
+
+```js
+// carbon-react/lib/components/__internal__/I18nContext/index.js
+import {createContext} from "react";
+
+export default createContext(() => {throw new Error("No I18nProvider exists.")});
+```
+
+```js
+// carbon-react/lib/components/I18nProvider/index.js
+import React from "react";
+import Context from "../__internal__/I18nContext";
+export default ({t, children}) => <Context.Provider value={t}>{children}</Context.Provider>
+```
+
+```js
+// carbon-react/lib/components/Example/index.js
+import React, { Suspense, useContext } from "react";
+import Context from "../__internal__/I18nContext";
+export default ({nextLabel}) => {
+  const t = useContext(Context);
+  return <span>{nextLabel || t("example.next")}</span>;
+};
+```
+
+Below is an example of how someone could configure lazy loading of carbon translations.
+This file is accessible from `/locales/fr-FR/carbon.json`
+
+```json
+{
+  "example": {
+    "next": "Suivant",
+    "previous": "Précédent"
+  }
+}
+```
+
+This file is accessible from `/locales/fr-FR/app.json`
+```json
+{
+  "example": {
+    "next": "Dernière page"
+  }
+}
+```
+
+```js
+// App.js
+import React, { Suspense } from "react";
+import HttpApi from "i18next-http-backend";
+import i18n from "i18next";
+import { initReactI18next, useTranslation } from "react-i18next";
+import Example from "carbon-react/lib/components/Example";
+import I18nProvider from "carbon-react/lib/components/I18nProvider";
+
+i18n
+  .use(initReactI18next)
+  .use(HttpApi)
+  .init({
+    interpolation: {
+      escapeValue: false,
+    },
+    lng: "fr-FR",
+    ns: ["carbon", "app"],
+    backend: {
+      loadPath: "/locales/{{lng}}/{{ns}}.json",
+    },
+    interpolation: {
```

Comment:
Why does it need the interpolation object twice?

Reply:
Good spot :)